YololCompetition/Services/Messages/DBMessages.cs

```diff
@@ -0,0 +1,106 @@
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+            }
+        }
+
+        public async Task TrackMessage(ulong ChannelID, ulong MessageID, ulong ChallengeID, uint MessageType)
+        {
+            await using var cmd = _database.CreateCommand();
+            cmd.CommandText = "INSERT into Messages (ChannelID, MessageID, ChallengeID, MessageType) values(@ChannelID, @MessageID, @ChallengeID, @MessageType)";
+            cmd.Parameters.Add(new SqliteParameter("@ChannelID", DbType.UInt64) { Value = ChannelID });
+            cmd.Parameters.Add(new SqliteParameter("@MessageID", DbType.UInt64) { Value = MessageID });
+            cmd.Parameters.Add(new SqliteParameter("@ChallengeID", DbType.UInt64) { Value = ChallengeID });
+            cmd.Parameters.Add(new SqliteParameter("@MessageType", DbType.UInt64) { Value = MessageType });
+            await cmd.ExecuteNonQueryAsync();
+        }
+
+        public IAsyncEnumerable<Message> GetCurrentMessages(ulong challengeID)
+        {
+            DbCommand PrepareQuery(IDatabase database)
+            {
+                var cmd = _database.CreateCommand();
+                cmd.CommandText = "SELECT from 'Messages' WHERE 'ChallengID' = @challengeID";
+                cmd.Parameters.Add(new SqliteParameter("@ChallengeID", DbType.UInt64) { Value = challengeID});
+                return cmd;
+            }
+
+            return new SqlAsyncResult<Message>(_database, PrepareQuery, ParseMessage);
+        }
+
+        private static Message ParseMessage(DbDataReader reader)
+        {
+            return new Message(
+                ulong.Parse(reader["ChannelID"].ToString()!),
+                ulong.Parse(reader["MessageID"].ToString()!),
+                ulong.Parse(reader["ChallengeID"].ToString()!),
+                uint.Parse(reader["MessageType"].ToString()!)
+            );
+        }
+
+        public void StartMessageWatch()
+        {
+            _cron.Schedule(TimeSpan.FromMinutes(5), TimeSpan.FromMinutes(1), uint.MaxValue, async () => {
+                // Get current challenge
+                var c = await _challenges.GetCurrentChallenge();
+
+                var messages = GetCurrentMessages(c.Id);
+                await foreach (Message entry in messages) 
+                {
+                    
+                    if (entry.ChallengeID != c.Id) {continue;}
+
+                    switch (entry.MessageType) {
+                        default:
+                        case 0:
+                        var channel = _client.GetChannel(entry.ChannelID);
+                        var chn = channel as ISocketMessageChannel;
+                        IUserMessage msg = chn.GetMessageAsync(entry.MessageID) as IUserMessage; 
+
+                        await msg.ModifyAsync(a => a.Embed = c.ToEmbed().Build());
```

Comment:
Add some extra null checks in case channel or message is null:

```
var channel = _client.GetChannel(entry.ChannelID) as ISocketMessageChannel;
IUserMessage msg = chn?.GetMessageAsync(entry.MessageID) as IUserMessage;
if (msg != null)
    await msg.ModifyAsync(a => a.Embed = c.ToEmbed().Build());
```